docs: refactor style guide example 07-04

danielsogl opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
Style guide example 07-04 uses the old Angular Http module.

## What is the new behavior?
Updated the example to use thew new HttpClient.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

danielsogl commented:
cc @brandonroberts

brandonroberts commented on aio/content/examples/styleguide/src/07-04/app/heroes/shared/hero.service.ts:
```diff
@@ -1,7 +1,6 @@
 import { Injectable } from '@angular/core';
-
-import { of } from 'rxjs';
+import { of } from 'rxjs/observable/of';
```
This import should be from 'rxjs'

danielsogl replied:
You are right!

danielsogl commented:
@brandonroberts fixed

gkalpak commented on aio/content/examples/styleguide/src/07-04/app/heroes/shared/hero-arena.service.avoid.ts:
```diff
@@ -1,14 +1,15 @@
   constructor(
       @Inject(HeroService) private heroService: HeroService,
-      @Inject(Http) private http: Http) {}
+      @Inject(Http) private http: HttpClient) {}
```
`@Inject(Http)` should be `@Inject(HttpClient)`.

danielsogl replied:
Makes sense.

danielsogl committed:
docs: refactor style guide example 07-04


docs: refactor style guide example 07-04


docs: refactor style guide example 07-04

danielsogl commented:
@gkalpak fixed

danielsogl commented:
@brandonroberts can you remove the pr-cleanup label?

brandonroberts reviewed (changes requested):
The `app.module.ts` in the `styleguide` folder needs to be updated to use `HttpClientModule` also.

brandonroberts reviewed (changes requested):
Remove all references to `@angular/http`

danielsogl commented:
I try will do the changes later this day

brandonroberts commented:
@danielsogl reopen this PR if you are still interested in landing this change.
